rusoto/core/src/region.rs

```diff
@@ -174,4 +197,49 @@ mod tests {
         assert_eq!(Region::UsWest2.to_string(), "us-west-2".to_owned());
         assert_eq!(Region::CnNorth1.to_string(), "cn-north-1".to_owned());
     }
+
+    #[test]
+    fn region_serialize_deserialize() {
+        assert_tokens(&Region::ApNortheast1, &[Token::String("ap-northeast-1")]);
+        assert_tokens(&Region::ApNortheast2, &[Token::String("ap-northeast-2")]);
+        assert_tokens(&Region::ApSouth1, &[Token::String("ap-south-1")]);
+        assert_tokens(&Region::ApSoutheast1, &[Token::String("ap-southeast-1")]);
+        assert_tokens(&Region::ApSoutheast2, &[Token::String("ap-southeast-2")]);
+        assert_tokens(&Region::CaCentral1, &[Token::String("ca-central-1")]);
+        assert_tokens(&Region::EuCentral1, &[Token::String("eu-central-1")]);
+        assert_tokens(&Region::EuWest1, &[Token::String("eu-west-1")]);
+        assert_tokens(&Region::EuWest2, &[Token::String("eu-west-2")]);
+        assert_tokens(&Region::SaEast1, &[Token::String("sa-east-1")]);
+        assert_tokens(&Region::UsEast1, &[Token::String("us-east-1")]);
+        assert_tokens(&Region::UsEast2, &[Token::String("us-east-2")]);
+        assert_tokens(&Region::UsWest1, &[Token::String("us-west-1")]);
+        assert_tokens(&Region::UsWest2, &[Token::String("us-west-2")]);
+        assert_tokens(&Region::CnNorth1, &[Token::String("cn-north-1")]);
+    }
+
+    #[test]
+    fn region_serialize_custom() {
+        let custom_region = Region::Custom("http://localhost:8000".to_owned());
+        assert_ser_tokens(&custom_region, &[Token::String("http://localhost:8000")]);
+    }
+
+    // Test we can still deserialize from the old style to ease migration
```

Comment:
Maybe we should also write a test to verify we can also deserialize from the normal AWS Regions? E.g. `ap-southeast-1`? That way whenever we do (if we do) drop the old style we still are testing that deserialization works

Reply:
the `assert_tokens` in the `region_serialize_deserialize` test assert both serialize and deserialization ([relavent docs](https://serde.rs/unit-testing.html)). 

`assert_ser_tokens` and `assert_de_tokens` are testing only individual directions, and only used for the functionality that currently only works in one direction (e.g. serializing custom regions, and deserializing from the `PascalCase` names like `UsEast1`)

Reply:
Huh! Well TIL, thanks for teaching me